bareos/director.sls

```diff
@@ -29,19 +29,36 @@ install_director_plugins:
       - pkgrepo: bareos_repo
     {% endif %}
 
+cleanup_director_default_config:
+  file.directory:
```

Comment:
If a user of the formula wants to use the default config provided by the packages (not defining `bareos.director.config` pillar data), this will make the formula fail, as it's removing the existing config and not adding new one.

You can see this happening in the `bareos-mysql-bento-ubuntu-1604` kitchen test, which fails because it removed the config file and did not add another one.

I'd suggest that this resource should be part of the conditional

```yaml
 {% if dir_config != {} %}
cleanup_director_default_config:
...
```
so default config is removed only if we take care of providing a new config file.